hudi-common/src/main/java/org/apache/hudi/common/model/HoodieWriteStat.java

```diff
@@ -114,6 +119,10 @@ public class HoodieWriteStat extends HoodieReadStats {
   @Nullable
   private RuntimeStats runtimeStats;
 
+  @JsonIgnore
```

Comment:
The annotation was not on the [HoodieDeltaWriteStat.java](https://github.com/apache/hudi/pull/13307/files#diff-d4fa29e0b797c44385dccd4fe35b8f36f120a786c522837a4a3305a8c5001774) Will this have side-effects?

Reply:
HoodieDeltaWriteStat does not have recordsStats field. I am rebasing this PR on top of HUDI-9413 again to bring in the latest changes.